### PR TITLE
Adapt postprocess to use new meta

### PR DESCRIPTION
**Pull request by tvlenin**

### CODE REVIEW

File: gst-libs/gst/r2inference/gstinferencepostprocess.c [hunk 1]
```diff
@@ -331,7 +332,8 @@ gst_create_class_from_prediction (GstVideoInference * vi,
 
 static void
 gst_get_boxes_from_prediction_float (gfloat obj_thresh, gfloat prob_thresh,
-    gpointer prediction, BBox * boxes, gint * elements, gint total_boxes)
+    gpointer prediction, BBox * boxes, gint * elements, gint total_boxes,
+    gdouble * probabilities)
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/gstinferencepostprocess.c]: Missing pointer guard. (Same as in yolov2)

File: gst-libs/gst/r2inference/gstinferencepostprocess.c [hunk 2]
```diff
@@ -361,6 +363,7 @@ gst_get_boxes_from_prediction_float (gfloat obj_thresh, gfloat prob_thresh,
       /* Iterate each class probability */
       for (c = 0; c < classes; c++) {
         cur_class_prob = ((gfloat *) prediction)[box_class_base + c];
+        probabilities[c] = cur_class_prob;
```
Comment (michaelgruner) [inline gst-libs/gst/r2inference/gstinferencepostprocess.c]: This overrides the probabilities of previous boxes with the new ones. Basically there’s a single vector of probabilities shared among all boxes.

File: gst-libs/gst/r2inference/gstinferencepostprocess.c [hunk 3]
```diff
@@ -26,7 +26,8 @@
 #define TOTAL_BOXES_5 845
 #define TOTAL_BOXES_15 2535
 
-#define TOTAL_CLASSES 20
+#define TOTAL_CLASSES_TINYYOLOV2 20
+#define TOTAL_CLASSES_TINYYOLOV3 80
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/r2inference/gstinferencepostprocess.c]: We shouldn't have definitions for specific models here. This should be refactored and the postprocess function of each element should configure the number of classes dynamically.

File: gst-libs/gst/r2inference/gstinferencepostprocess.h [hunk 4]
```diff
@@ -67,7 +68,8 @@ gboolean gst_create_boxes (GstVideoInference * vi, const gpointer prediction,
 gboolean gst_create_boxes_float (GstVideoInference * vi,
     const gpointer prediction, gboolean * valid_prediction,
     BBox ** resulting_boxes, gint * elements, gdouble obj_thresh,
-    gdouble prob_thresh, gdouble iou_thresh);
+    gdouble prob_thresh, gdouble iou_thresh, gdouble ** probabilities,
+    gint num_classes);
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/r2inference/gstinferencepostprocess.h]: Update above documentation with the new parameters. Try to never leave documentation behind!

File: gst-libs/gst/r2inference/gstinferencepostprocess.h [hunk 5]
```diff
@@ -50,7 +50,8 @@ G_BEGIN_DECLS
  */
 gboolean gst_create_boxes (GstVideoInference * vi, const gpointer prediction,
     gboolean * valid_prediction, BBox ** resulting_boxes,
-    gint * elements, gfloat obj_thresh, gfloat prob_thresh, gfloat iou_thresh, gdouble ** probabilities);
+    gint * elements, gfloat obj_thresh, gfloat prob_thresh, gfloat iou_thresh,
+    gdouble ** probabilities, gint num_classes);
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/r2inference/gstinferencepostprocess.h]: Update above documentation with the new parameters. Try to never leave documentation behind!

**Pull request merged**